### ISSUE TITLE
code: remove slog dependency

### ISSUE THREAD

**Issue by alixander** [terrastruct/d2#1637]

**Comment by lukasbindreiter** [comment]
@alixander 

Is this planned to be implemented in the near future?

Because especially when using `d2` as a library in another go project this is a bit of an issue already - since d2 is only compatible with `cdr.dev/slog <= 1.5`, the latest `slog` version has some backward incompatible changes:

```
# oss.terrastruct.com/d2/lib/log
../../../go/pkg/mod/oss.terrastruct.com/d2@v0.6.7/lib/log/log.go:52:28: cannot use fields (variable of type []slog.Field) as []any value in argument to from(ctx).Debug
../../../go/pkg/mod/oss.terrastruct.com/d2@v0.6.7/lib/log/log.go:57:27: cannot use fields (variable of type []slog.Field) as []any value in argument to from(ctx).Info
../../../go/pkg/mod/oss.terrastruct.com/d2@v0.6.7/lib/log/log.go:62:27: cannot use fields (variable of type []slog.Field) as []any value in argument to from(ctx).Warn
../../../go/pkg/mod/oss.terrastruct.com/d2@v0.6.7/lib/log/log.go:67:28: cannot use fields (variable of type []slog.Field) as []any value in argument to from(ctx).Error
../../../go/pkg/mod/oss.terrastruct.com/d2@v0.6.7/lib/log/log.go:72:31: cannot use fields (variable of type []slog.Field) as []any value in argument to from(ctx).Critical
../../../go/pkg/mod/oss.terrastruct.com/d2@v0.6.7/lib/log/log.go:77:28: cannot use fields (variable of type []slog.Field) as []any value in argument to from(ctx).Fatal
```

This means whenever I ugrade dependencies in my project, I typically do:
```
go get -u ./...
// then manually edit go.mod and revert cdr.dev/slog to 1.5 otherwise it won't compile anymore
go build ./...
// and then do some testing to see everything works
```

Go now even offers a `slog` package in the standard libary:
https://go.dev/blog/slog

I think it would make lot's of sense to adapt this instead.

**Comment by lukasbindreiter** [comment]
@alixander 

I just realized it's even more of an issue than what I thought, there seems to be something breaking this with `d2` v0.6.7.

Steps:
1. create an empty new go project
2. run `go get oss.terrastruct.com/d2`
3. this will download and add `d2 v0.6.7` and `cdr.dev/slog v1.6.1` to the project, *which won't compile*
4. manually downgrading `slog` to `v1.5.0` won't work, as soon as you run `go get` it upgrades it again
5. So I have to both pin terrastruct/d2 to `v0.6.6` and `cdr.dev/slog` to `v1.5.0`

I've noticed that you have a `replace` directive in place: https://github.com/terrastruct/d2/blob/8223e860f17ffb12acfe65bcbe9dfd0c4251e2b6/go.mod#L7

However, i want to avoid this in our case, it seems a bit hacky.
